src/Tribe/Integrations/Freemius.php

```diff
@@ -0,0 +1,386 @@
+		$freemius = tribe( 'freemius' );
+
+		$this->instance = $freemius->initialize( $this->slug, $this->freemius_id, $this->public_key, [
+			'menu'           => [
+				'slug'       => $this->page,
+				'first-path' => $this->get_settings_path(),
+				'account'    => false,
+				'contact'    => false,
+				'support'    => false,
+			],
+			'is_premium'     => false,
+			'has_addons'     => false,
+			'has_paid_plans' => false,
+		] );
+
+		$this->instance->add_filter( 'connect_url', [ $this, 'get_connect_url' ], 10, 10 );
+		$this->instance->add_filter( 'after_skip_url', [ $this, 'get_settings_url' ] );
+		$this->instance->add_filter( 'after_connect_url', [ $this, 'get_settings_url' ] );
+		$this->instance->add_filter( 'after_pending_connect_url', [ $this, 'get_settings_url' ] );
+		$this->instance->add_filter( 'plugin_icon', [ $this, 'get_plugin_icon_url' ] );
+
+		/*
+		 * Freemius typically hooks this action–which bootstraps the deactivation dialog–during plugins_loaded, but we
+		 * initialize our plugins AFTER plugins_loaded, so we'll register it on admin_init instead.
+		 */
+		add_action( 'admin_init', [ $this->instance, '_hook_action_links_and_register_account_hooks' ] );
+		add_action( 'admin_init', [ $this, 'action_skip_activation' ] );
+
+		$this->instance->add_filter( 'connect_message', [
+			$this,
+			'filter_connect_message_on_update',
+		], 10, 6 );
+		$this->instance->add_filter( 'connect_message_on_update', [
+			$this,
+			'filter_connect_message_on_update',
+		], 10, 6 );
+
+		add_action( 'admin_init', [ $this, 'maybe_remove_activation_complete_notice' ] );
+	}
+
+	/**
+	 * For some reason Freemius is redirecting some customers to a page that doesnt exist. So we catch that page and
+	 * redirect them back to the actual page that we are using to setup the plugins integration.
+	 *
+	 * @since TBD
+	 *
+	 * @link https://moderntribe.atlassian.net/browse/TEC-3218
+	 *
+	 * @return void  Retuning a Redirect header, so nothing gets returned otherwise.
+	 */
+	public function action_redirect_incorrect_page() {
+		$action = tribe_get_request_var( 'fs_action', false );
+
+		if ( 'sync_user' !== $action ) {
+			return;
+		}
+
+		$page = tribe_get_request_var( 'page', false );
+
+		if ( 'tribe-common-account' !== $page ) {
+			return;
+		}
+
+		$url = admin_url( 'admin.php' );
+		$url = add_query_arg( [
+			'fs_action' => $action,
+			'page'      => $this->page,
+			'_wpnonce'  => tribe_get_request_var( '_wpnonce' ),
+		], $url );
+
+		wp_safe_redirect( $url );
+		tribe_exit();
+	}
+
+	/**
+	 * Get the connect page URL.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $connect_url Current connect page URL.
+	 *
+	 * @return string The connect page URL.
+	 */
+	public function get_connect_url( $connect_url ) {
+		$settings_url = $this->get_settings_url();
+
+		if ( false !== strpos( $connect_url, 'fs_action' ) ) {
+			$action = $this->slug . '_reconnect';
+
+			$settings_url = add_query_arg( [
+				'nonce'     => wp_create_nonce( $action ),
```

Comment:
:no_entry_sign: Opening parenthesis of a multi-line function call must be the last content on the line

<sub>hash: adeb54ff5f1c8e9e3c56a7d8b001bd02</sub>